### PR TITLE
Adds an error when the dataform.json file is invalid

### PR DESCRIPTION
**Pull request by GJMcGowan**
Working towards solving #1272

### CODE REVIEW

File: core/gen_index.ts [hunk 1]
```diff
@@ -32,12 +32,23 @@ export function genIndex(base64EncodedConfig: string): string {
     })
     .join("\n");
 
+  const safeIncludeDataformJson = `
```
Comment (BenBirt) [inline core/gen_index.ts]: can we instead just do a JSON parse of dataform.json, in native typescript code?

File: api/commands/compile.ts [hunk 2]
```diff
@@ -10,6 +10,17 @@ export async function compile(
 ): Promise<dataform.CompiledGraph> {
   // Resolve the path in case it hasn't been resolved already.
   path.resolve(compileConfig.projectDir);
+
+  try {
+    // check dataformJson is valid before we try to compile
+    const dataformJson = fs.readFileSync(`${compileConfig.projectDir}/dataform.json`, "utf8");
+    JSON.parse(dataformJson);
+  } catch (e) {
+    const error = new Error("Compile Error: `dataform.json` is invalid");
```
Comment (BenBirt) [inline api/commands/compile.ts]: this stacktrace is probably a lot cleaner... can we just `throw new Error("Contents of 'dataform.json' file are invalid.");`?

File: version.bzl [hunk 3]
```diff
@@ -1,3 +1,3 @@
 # NOTE: If you change the format of this line, you must change the bash command
 # in /scripts/publish to extract the version string correctly.
-DF_VERSION = "1.0.9"
+DF_VERSION = "1.0.10"
```
Comment (BenBirt) [inline version.bzl]: no need for this now.

Comment (GJMcGowan) [reply version.bzl]: Isn't there? If I want to require this in `dataform-co` do I not refer to the version number? I guess I don't really understand the release process for this.

**Pull request merged**